src/definitions/modules/calendar.js

```diff
@@ -802,24 +802,64 @@
+                        if (!module.helper.sanitiseDate($module.data(metadata.date))) {
+                            return null;
+                        }
+
+                        if (format === '') {
+                            return module.helper.sanitiseDate($module.data(metadata.date));
+                        }
+
+                        return module.helper.dateFormat(format, $module.data(metadata.date));
                     },
                     inputDate: function () {
                         return $input.val();
                     },
-                    focusDate: function () {
-                        return $module.data(metadata.focusDate) || null;
+                    focusDate: function (format = '') {
+                        if (!module.helper.sanitiseDate($module.data(metadata.focusDate))) {
+                            return null;
+                        }
+
+                        if (format === '') {
+                            return module.helper.sanitiseDate($module.data(metadata.focusDate));
+                        }
+
+                        return module.helper.dateFormat(format, $module.data(metadata.focusDate));
                     },
-                    startDate: function () {
+                    startDate: function (format = '') {
                         var startModule = module.get.calendarModule(settings.startCalendar);
 
-                        return (startModule ? startModule.get.date() : $module.data(metadata.startDate)) || null;
+                        if (startModule) {
+                            return startModule.get.date(format);
+                        }
+
+                        if (!module.helper.sanitiseDate($module.data(metadata.startDate))) {
+                            return null;
+                        }
+
+                        if (format === '') {
+                            return module.helper.sanitiseDate($module.data(metadata.startDate));
+                        }
+
+                        return module.helper.dateFormat(format, $module.data(metadata.startDate));
                     },
-                    endDate: function () {
+                    endDate: function (format = '') {
                         var endModule = module.get.calendarModule(settings.endCalendar);
 
-                        return (endModule ? endModule.get.date() : $module.data(metadata.endDate)) || null;
+                        if (endModule) {
+                            return endModule.get.date(format);
+                        }
+
+                        if (!module.helper.sanitiseDate($module.data(metadata.endDate))) {
+                            return null;
+                        }
+
+                        if (format === '') {
+                            return module.helper.sanitiseDate($module.data(metadata.endDate));
+                        }
+
+                        return module.helper.dateFormat(format, $module.data(metadata.endDate));
```

Comment:
1. Default function parameter values are not supported in IE11 (which FUI 2.9.x still supports)
2. I would reduced the call to the sanitisedate function
```suggestion
                        format = format || '';
                        if (endModule) {
                            return endModule.get.date(format);
                        }
                        
                        var endDate = module.helper.sanitiseDate($module.data(metadata.endDate)) || null;
                        if (!endDate) {
                            return null;
                        }

                        if (format === '') {
                            return endDate;
                        }

                        return module.helper.dateFormat(format, endDate);
```